mypy/plugins/attrs.py

```diff
@@ -967,16 +964,25 @@ def evolve_function_sig_callback(ctx: mypy.plugin.FunctionSigContext) -> Callabl
 
     inst_type = get_proper_type(inst_type)
     if isinstance(inst_type, AnyType):
-        return ctx.default_signature
+        return ctx.default_signature  # evolve(Any, ....) -> Any
     inst_type_str = format_type_bare(inst_type)
-
-    attrs_init_type = _get_attrs_init_type(inst_type)
-    if not attrs_init_type:
+    attrs_type = get_proper_type(
+        inst_type.upper_bound if isinstance(inst_type, TypeVarType) else inst_type
+    )
+    attrs_init_type = None
+    if isinstance(attrs_type, Instance):
+        attrs_init_type = _get_attrs_init_type(attrs_type)
+    if attrs_init_type is None:
         ctx.api.fail(
-            f'Argument 1 to "evolve" has incompatible type "{inst_type_str}"; expected an attrs class',
+            f'Argument 1 to "evolve" has a variable type "{inst_type_str}" not bound to an attrs class'
```

Comment:
Should this be more friendly, e.g.

> has a variable type "T" that is not necessarily an attrs class; consider adding bounds=<attrs-class> to your TypeVar definition

I can't tell where we stand between succinct and instructive.